### PR TITLE
Increase tests coverage up to 100%

### PR DESCRIPTION
**Pull request by pwilczynskiclearcode**
- adds missing tests
- minor fixes to conform with pylint rules

Even library covers 100% of code it doesn't cover 100% of use cases!

### CODE REVIEW

**Comment by pwilczynskiclearcode** [comment]
Latest coverage https://coveralls.io/builds/2119440 - 100%

**Comment by fizyk** [comment]
:+1:

File: mirakuru/output.py [hunk 1]
```diff
@@ -45,6 +45,7 @@ def __init__(self, command, banner, **kwargs):
         """
         super(OutputExecutor, self).__init__(command, **kwargs)
         self._banner = re.compile(banner)
+        self.poll_obj = None
```
Comment (tomislater) [inline mirakuru/output.py]: Hmm, what is it? I mean, we add this in ``start`` method.

File: mirakuru/base.py [hunk 2]
```diff
@@ -236,9 +245,7 @@ def check_timeout(self):
         :returns: True if timeout expired, False if not
         :rtype: bool
         """
-        if self._endtime is not None and time.time() > self._endtime:
-            return False
-        return True
+        return self._endtime is None or time.time() <= self._endtime
```
Comment (tomislater) [inline mirakuru/base.py]: :+1:

**Comment by tomislater** [comment]
:+1:

File: mirakuru/output.py (hunk 1 above)
Comment (pwilczynskiclearcode) [inline mirakuru/output.py]: Well that's the `pylint` good practice: not to introduce new instance attributes during lifetime of an object. It helps with unit-tests (you can mock some attribute before a method test) and understanding of a model is easier (you know what entire class consists of without a need of each method studies).

**Pull request merged**